src/util.js

```diff
@@ -55,29 +25,42 @@ exports.serialize = (dagNode, callback) => {
   callback(null, serialized)
 }
 
-exports.deserialize = (data, callback) => {
+function deserialize (data, callback) {
   const pbn = proto.PBNode.decode(data)
 
   const links = pbn.Links.map((link) => {
     return new DAGLink(link.Name, link.Tsize, link.Hash)
   })
 
-  stable.inplace(links, exports.linkSort)
-
   const buf = pbn.Data || new Buffer(0)
 
-  const dagNode = new DAGNode(buf, links)
-
-  callback(null, dagNode)
+  DAGNode.create(buf, links, callback)
```

Comment:
@diasdavid we talked about the IPLD format API and this is pretty much what I meant how it forces sub-optimal code: deserialize depends on DAGNode and DAGNode depends on serialize (serialize and deserialize being both in the same file) essentially creating a cyclic dependency between the two. For the sake of getting this shipped, let's not focus on that but keep this in mind for future discussions.